spring-kafka/src/main/java/org/springframework/kafka/core/KafkaAdmin.java

```diff
@@ -0,0 +1,205 @@
+
+	public void setRunning(boolean running) {
+		this.running = running;
+	}
+
+	public void setPhase(int phase) {
+		this.phase = phase;
+	}
+
+	public void setAutoStartup(boolean autoStartup) {
+		this.autoStartup = autoStartup;
+	}
+
+	/**
+	 * Set the close timeout in seconds. Defaults to 10 seconds.
+	 * @param closeTimeout the timeout.
+	 */
+	public void setCloseTimeout(int closeTimeout) {
+		this.closeTimeout = closeTimeout;
+	}
+
+	@Override
+	public synchronized void start() {
+		if (!this.running) {
+			if (this.applicationContext != null) {
+				addTopicsIfNeeded(this.applicationContext.getBeansOfType(NewTopic.class, false, false).values());
+				addAclBindingsIfNeeded(
+						this.applicationContext.getBeansOfType(AclBinding.class, false, false).values());
+			}
+			this.running = true;
+		}
+	}
+
+	@Override
+	public synchronized void stop() {
+		if (this.running) {
+			this.running = false;
+		}
+	}
+
+	@Override
+	public boolean isRunning() {
+		return this.running;
+	}
+
+	@Override
+	public int getPhase() {
+		return this.phase;
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return this.autoStartup;
+	}
+
+	@Override
+	public void stop(Runnable callback) {
+		stop();
+		callback.run();
+	}
+
+	@Override
+	public void destroy() throws Exception {
+		this.adminClient.close(this.closeTimeout, TimeUnit.SECONDS);
+	}
+
+	private void addTopicsIfNeeded(Collection<NewTopic> topics) {
+		if (topics.size() > 0) {
+			Map<String, NewTopic> topicNameToTopic = new HashMap<>();
+			topics.stream().forEach(t -> topicNameToTopic.compute(t.name(), (k, v) -> v = t));
+			DescribeTopicsResult topicInfo = this.adminClient
+					.describeTopics(topics.stream().map(t -> t.name()).collect(Collectors.toList()));
+			List<NewTopic> topicsToAdd = new ArrayList<>();
+			topicInfo.values().forEach((n, f) -> {
+				try {
+					TopicDescription topicDescription = f.get();
+					if (topicNameToTopic.get(n).numPartitions() != topicDescription.partitions().size()) {
+						if (logger.isDebugEnabled()) {
+							logger.debug(String.format(
+									"Topic '%s' exists but has a different partition count: %d not %d", n,
+									topicDescription.partitions().size(), topicNameToTopic.get(n).numPartitions()));
+						}
+					}
+				}
+				catch (InterruptedException e) {
+					Thread.currentThread().interrupt();
+				}
+				catch (ExecutionException e) {
+					topicsToAdd.add(topicNameToTopic.get(n));
+				}
+			});
+			if (topicsToAdd.size() > 0) {
+				CreateTopicsResult topicResults = this.adminClient.createTopics(topicsToAdd);
+				try {
+					topicResults.all().get();
+				}
+				catch (InterruptedException e) {
+					Thread.currentThread().interrupt();
+					logger.error("Interrupted while waiting for topic creation results", e);
+				}
+				catch (ExecutionException e) {
+					logger.error("Failed to create topics", e.getCause());
+				}
+			}
+		}
+	}
+
+	private void addAclBindingsIfNeeded(Collection<AclBinding> bindings) {
+		// TODO (or not)
```

Comment:
I would leave it without any ACL auto-binding. That looks like rare task and would be possible via `AdminClient`.
Creating topics by the application context is really something people like from Spring AMQP.